tests/testDatabase.js

```diff
@@ -0,0 +1,149 @@
+var ERR = require('async-stacktrace');
+var async = require('async');
+var pg = require('pg');
+var assert = require('chai').assert;
+var colors = require('colors');
+
+var databaseDiff = require('./databaseDiff');
+var migrations = require('../migrations');
+var sqldb = require('../lib/sqldb');
+var sqlLoader = require('../lib/sql-loader');
+var sql = sqlLoader.loadSqlEquiv(__filename);
+
+var postgresqlUser = 'postgres';
+var postgresqlHost = 'localhost';
+var initConString = 'postgres://localhost/postgres';
+
+
+// Custom error type so we can handle this specially
+function DatabaseError(message) {
+  this.name = 'DatabaseError';
+  this.message = message;
+}
+DatabaseError.prototype = Object.create(Error.prototype);
+DatabaseError.prototype.constructor = DatabaseError;
+
+describe('database', function() {
+
+    it('should match the database described in /database', function(done) {
+        this.timeout(10000);
+        let results = '';
+        let errMsg = '';
+        async.series([
+            (callback) => setupNamedDatabase('migrations_test', callback),
+            (callback) => {
+                const options = {
+                    outputFormat: 'string',
+                    coloredOutput: process.stdout.isTTY,
+                };
+                databaseDiff.diffDirectoryAndDatabase('database', 'migrations_test', options, (err, data) => {
+                    if (ERR(err, callback)) return;
+                    errMsg = data;
+                    callback(null);
+                });
+            },
+            (callback) => tearDownNamedDatabase('migrations_test', callback),
+        ], (err) => {
+            if (ERR(err, done)) return;
+            if (errMsg) {
+                // We add a newline with red ANSI codes so that mocha's default
+                // red coloring doesn't mess up our diff rendering
+                done(new DatabaseError('\n'.red + errMsg));
+            } else {
+                done(null);
+            }
+        });
+    });
+});
+
+function setupNamedDatabase(databaseName, callback) {
```

Comment:
Could we replace most of the `before()` and `after()` functions in `helperDb.js` with calls to `setupNamedDatabase()` and `tearDownNamedDatabase()`?